test/v2/controllers/validators/RetrieveUkPropertyPeriodSummaryValidatorFactorySpec.scala

```diff
@@ -0,0 +1,121 @@
+ */
+
+package v2.controllers.validators
+
+import api.models.domain.{BusinessId, Nino, SubmissionId, TaxYear}
+import api.models.errors._
+import mocks.MockAppConfig
+import support.UnitSpec
+import v2.models.request.retrieveUkPropertyPeriodSummary.RetrieveUkPropertyPeriodSummaryRequestData
+
+class RetrieveUkPropertyPeriodSummaryValidatorFactorySpec extends UnitSpec with MockAppConfig {
+  private implicit val correlationId: String = "1234"
+
+  private val validNino         = "AA123456A"
+  private val validBusinessId   = "XAIS12345678901"
+  private val validTaxYear      = "2022-23"
+  private val validSubmissionId = "4557ecb5-fd32-48cc-81f5-e6acd1099f3c"
+
+  private val parsedNino         = Nino(validNino)
+  private val parsedBusinessId   = BusinessId(validBusinessId)
+  private val parsedTaxYear      = TaxYear.fromMtd(validTaxYear)
+  private val parsedSubmissionId = SubmissionId(validSubmissionId)
+
+  private val validatorFactory = new RetrieveUkPropertyPeriodSummaryValidatorFactory(mockAppConfig)
+
+  private def validator(nino: String, businessId: String, taxYear: String, submissionId: String) =
+    validatorFactory.validator(nino, businessId, taxYear, submissionId)
+
+  class SetUp {
+
+    MockAppConfig.minimumTaxV2Uk
+      .returns(2022)
+      .anyNumberOfTimes()
+
+  }
+
+  "validator" should {
+    "return the parsed domain object" when {
+      "passed a valid request" in new SetUp {
+        val result: Either[ErrorWrapper, RetrieveUkPropertyPeriodSummaryRequestData] =
+          validator(validNino, validBusinessId, validTaxYear, validSubmissionId).validateAndWrapResult()
+
+        result shouldBe Right(RetrieveUkPropertyPeriodSummaryRequestData(parsedNino, parsedBusinessId, parsedTaxYear, parsedSubmissionId))
+      }
+    }
+
+    "return a single error" when {
+      "an invalid nino is supplied" in new SetUp {
+        val result: Either[ErrorWrapper, RetrieveUkPropertyPeriodSummaryRequestData] =
+          validator("5555", validBusinessId, validTaxYear, validSubmissionId).validateAndWrapResult()
+
+        result shouldBe Left(ErrorWrapper(correlationId, NinoFormatError))
+      }
+
+      "an invalid business ID is supplied" in new SetUp {
+        val result: Either[ErrorWrapper, RetrieveUkPropertyPeriodSummaryRequestData] =
+          validator(validNino, "5555", validTaxYear, validSubmissionId).validateAndWrapResult()
+
+        result shouldBe Left(ErrorWrapper(correlationId, BusinessIdFormatError))
+      }
+
+      "an incorrectly formatted tax year is supplied" in new SetUp {
+        val result: Either[ErrorWrapper, RetrieveUkPropertyPeriodSummaryRequestData] =
+          validator(validNino, validBusinessId, "24333", validSubmissionId).validateAndWrapResult()
+
+        result shouldBe Left(ErrorWrapper(correlationId, TaxYearFormatError))
+      }
+
+      "an unsupported tax year is supplied" in new SetUp {
+        val result: Either[ErrorWrapper, RetrieveUkPropertyPeriodSummaryRequestData] =
```

Comment:
same here about using the last invalid taxYear (see comment on test/v2/controllers/validators/ListPropertyPeriodSummariesValidatorFactorySpec)